sdk/storage/CMakeLists.txt

```diff
@@ -10,30 +10,26 @@ set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 option(BUILD_STORAGE_SAMPLES "Build storage sample codes" ON)
 
+add_subdirectory(azure-storage-common)
+add_subdirectory(azure-storage-blobs)
+add_subdirectory(azure-storage-files-datalake)
+add_subdirectory(azure-storage-files-shares)
 
-if(MSVC)
-    add_compile_definitions(NOMINMAX)
-endif()
-
-
-add_executable(azure-storage-test)
 if(BUILD_TESTING)
-    set_target_properties(azure-storage-test PROPERTIES EXCLUDE_FROM_ALL FALSE)
+    add_executable(azure-storage-test)
     add_gtest(azure-storage-test)
-else()
-    set_target_properties(azure-storage-test PROPERTIES EXCLUDE_FROM_ALL TRUE)
-endif()
 
-
-add_executable(azure-storage-sample)
-if(BUILD_STORAGE_SAMPLES)
-    set_target_properties(azure-storage-sample PROPERTIES EXCLUDE_FROM_ALL FALSE)
-else()
-    set_target_properties(azure-storage-sample PROPERTIES EXCLUDE_FROM_ALL TRUE)
+    add_subdirectory(azure-storage-common/test)
+    add_subdirectory(azure-storage-blobs/test)
+    add_subdirectory(azure-storage-files-datalake/test)
+    add_subdirectory(azure-storage-files-shares/test)
```

Comment:
This looks like another candidate for creating an independent PR.
It looks like this is refactoring the Azure Storage tests and samples build structure, which makes me happy!, but should be committed as a different story